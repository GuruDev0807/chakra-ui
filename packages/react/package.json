{
  "name": "@chakra-ui/react",
  "version": "1.7.0",
  "description": "Responsive and accessible React UI components built with React and Emotion",
  "main": "dist/chakra-ui-react.cjs.js",
  "module": "dist/chakra-ui-react.esm.js",
  "types": "dist/chakra-ui-react.cjs.d.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@chakra-ui/accordion": "1.4.0",
    "@chakra-ui/alert": "1.3.0",
    "@chakra-ui/avatar": "1.3.0",
    "@chakra-ui/breadcrumb": "1.3.0",
    "@chakra-ui/button": "1.5.0",
    "@chakra-ui/checkbox": "1.6.0",
    "@chakra-ui/close-button": "1.2.0",
    "@chakra-ui/control-box": "1.1.0",
    "@chakra-ui/counter": "1.2.0",
    "@chakra-ui/css-reset": "1.1.0",
    "@chakra-ui/editable": "1.3.0",
    "@chakra-ui/form-control": "1.5.0",
    "@chakra-ui/hooks": "1.7.0",
    "@chakra-ui/icon": "1.2.0",
    "@chakra-ui/image": "1.1.0",
    "@chakra-ui/input": "1.3.0",
    "@chakra-ui/layout": "1.5.0",
    "@chakra-ui/live-region": "1.1.0",
    "@chakra-ui/media-query": "1.2.0",
    "@chakra-ui/menu": "1.8.0",
    "@chakra-ui/modal": "1.10.0",
    "@chakra-ui/number-input": "1.3.0",
    "@chakra-ui/pin-input": "1.7.0",
    "@chakra-ui/popover": "1.10.0",
    "@chakra-ui/popper": "2.4.0",
    "@chakra-ui/portal": "1.3.0",
    "@chakra-ui/progress": "1.2.0",
    "@chakra-ui/provider": "1.7.0",
    "@chakra-ui/radio": "1.4.0",
    "@chakra-ui/react-env": "1.1.0",
    "@chakra-ui/select": "1.2.0",
    "@chakra-ui/skeleton": "1.2.0",
    "@chakra-ui/slider": "1.5.0",
    "@chakra-ui/spinner": "1.2.0",
    "@chakra-ui/stat": "1.2.0",
    "@chakra-ui/switch": "1.3.0",
    "@chakra-ui/system": "1.8.0",
    "@chakra-ui/table": "1.3.0",
    "@chakra-ui/tabs": "1.6.0",
    "@chakra-ui/tag": "1.2.0",
    "@chakra-ui/textarea": "1.2.0",
    "@chakra-ui/theme": "1.12.0",
    "@chakra-ui/toast": "1.4.0",
    "@chakra-ui/tooltip": "1.4.0",
    "@chakra-ui/transition": "1.4.0",
    "@chakra-ui/utils": "1.9.0",
    "@chakra-ui/visually-hidden": "1.1.0"
  },
  "peerDependencies": {
    "@emotion/react": "^11.0.0",
    "@emotion/styled": "^11.0.0",
    "framer-motion": "3.x || 4.x || 5.x",
    "react": ">=16.8.6",
    "react-dom": ">=16.8.6"
  },
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "license": "MIT",
  "homepage": "https://chakra-ui.com/",
  "repository": {
    "type": "git",
    "url": "https://github.com/chakra-ui/chakra-ui",
    "directory": "packages/react"
  },
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "storybook": {
    "title": "Chakra UI",
    "url": "https://chakra-ui.netlify.app"
  },
  "devDependencies": {
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "framer-motion": "^4.0.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.2"
  }
}
