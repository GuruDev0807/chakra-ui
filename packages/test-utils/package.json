{
  "name": "@chakra-ui/test-utils",
  "version": "1.0.0-next.6",
  "private": true,
  "description": "Testing utilities for Chakra UI's components",
  "keywords": [
    "chakra",
    "test utils",
    "testing",
    "utils",
    "testing library",
    "react testing library"
  ],
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "homepage": "https://github.com/chakra-ui/chakra-ui#readme",
  "license": "MIT",
  "main": "src/index.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chakra-ui/chakra-ui.git"
  },
  "bugs": {
    "url": "https://github.com/chakra-ui/chakra-ui/issues"
  },
  "dependencies": {
    "@chakra-ui/system": "^1.0.0-next.6",
    "@chakra-ui/theme": "^1.0.0-next.6",
    "@testing-library/jest-dom": "5.1.1",
    "@testing-library/react": "10.4.5",
    "@testing-library/react-hooks": "3.2.1",
    "@testing-library/user-event": "12.0.11",
    "@types/jest-axe": "3.2.1",
    "jest-axe": "3.4.0",
    "react-test-renderer": "16.12.0"
  },
  "peerDependencies": {
    "react": "16.x"
  }
}
