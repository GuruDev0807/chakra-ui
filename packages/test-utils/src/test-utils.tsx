import { ChakraProvider } from "@chakra-ui/core"
import "@testing-library/jest-dom/extend-expect"
import {
  render as RTLRender,
  RenderOptions,
  fireEvent,
  RenderResult,
  Queries,
} from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations } from "jest-axe"
import serializer from "jest-emotion"

expect.addSnapshotSerializer(serializer)
expect.extend(toHaveNoViolations)

const AllProviders: React.FC = ({ children }) => (
  <ChakraProvider resetCSS>{children}</ChakraProvider>
)

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult

export function render(ui: React.ReactElement, options?: RenderOptions) {
  return RTLRender(ui, { wrapper: AllProviders, ...options })
}

export * from "@testing-library/react"

export {
  act as invoke,
  renderHook,
  RenderHookOptions,
  RenderHookResult,
} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export * from "jest-axe"

export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, { key: "Escape", keyCode: 27 })
