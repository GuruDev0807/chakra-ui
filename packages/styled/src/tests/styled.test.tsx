import * as React from "react"
import { render } from "@chakra-ui/test-utils"
import { chakra } from ".."

test("as jsx element", () => {
  const Div = chakra("div")
  const tools = render(<Div as="span">a</Div>)

  expect(tools.getByText("a")).toMatchInlineSnapshot(`
    <span>
      a
    </span>
  `)
})

test("as component", () => {
  type Props = {
    isBig?: boolean
  }
  const Konoha = ({ isBig, ...props }: Props) => (
    <div data-size={isBig ? "big" : "small"} {...props} />
  )

  const Div = chakra("div")

  const tools = render(
    <Div as={Konoha} isBig>
      a
    </Div>,
  )

  expect(tools.getByText("a")).toMatchInlineSnapshot(`
    <div
      data-size="big"
    >
      a
    </div>
  `)
})

test("chakra elements - renders currently", () => {
  const tools = render(<chakra.div>Welcome</chakra.div>)

  expect(tools.getByText("Welcome")).toMatchInlineSnapshot(`
    <div>
      Welcome
    </div>
  `)
})

test("chakra elements - renders with style props", () => {
  const tools = render(<chakra.div marginTop="40px">Welcome</chakra.div>)

  expect(tools.asFragment()).toMatchSnapshot()

  expect(tools.getByText("Welcome")).toHaveStyle(`margin-top: 40px`)
})

test("chakra function - renders currently", () => {
  const Button = chakra("button")
  const tools = render(<Button>Click Me</Button>)

  expect(tools.getByText("Click Me")).toMatchInlineSnapshot(`
    <button>
      Click Me
    </button>
  `)

  tools.rerender(
    <Button
      bg="blue.600"
      color="white"
      _hover={{ bg: "blue.700" }}
      _active={{ bg: "blue.800" }}
    >
      Click me
    </Button>,
  )

  expect(tools.asFragment()).toMatchSnapshot()
})

test("it allows pass through props", () => {
  const Image = chakra("img")
  const tools = render(<Image data-testid="img" htmlWidth="60px" />)

  expect(tools.asFragment()).toMatchSnapshot()
})
