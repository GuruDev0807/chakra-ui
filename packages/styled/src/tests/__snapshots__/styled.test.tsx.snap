// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attrs option works correctly 1`] = `
<DocumentFragment>
  <button
    type="button"
  >
    Click
  </button>
</DocumentFragment>
`;

exports[`chakra elements - renders with style props 1`] = `
<DocumentFragment>
  <div
    class="css-y1gt6f"
  >
    Welcome
  </div>
</DocumentFragment>
`;

exports[`chakra function - renders currently 2`] = `
<DocumentFragment>
  <button
    class="css-gnqum9"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`it allows pass through props 1`] = `
<DocumentFragment>
  <img
    data-testid="img"
    width="60px"
  />
</DocumentFragment>
`;

exports[`it renders component using theme key 1`] = `
<DocumentFragment>
  <span>
    Badge
  </span>
</DocumentFragment>
`;

exports[`it renders with base style 1`] = `
<DocumentFragment>
  <div
    class="css-pfs33m"
  >
    Hey!
  </div>
</DocumentFragment>
`;
